sdk/core/core-client-rest/src/sendRequest.ts

```diff
@@ -98,16 +99,30 @@ interface RequestBody {
 /**
  * Prepares the body before sending the request
  */
-function getRequestBody(body?: unknown, contentType: string = "application/json"): RequestBody {
+function getRequestBody(body?: unknown, contentType: string = ""): RequestBody {
   if (body === undefined) {
     return { body: undefined };
   }
 
+  if (!contentType && typeof body === "string") {
+    return { body };
+  }
+
   const firstType = contentType.split(";")[0];
 
+  if (firstType === "application/json") {
+    return { body: JSON.stringify(body) };
+  }
+
+  if (ArrayBuffer.isView(body)) {
+    return { body: binaryArrayToString(body) };
+  }
```

Comment:
would it be too messy if we split out the handling for non-Uint8Arrays here so that `binaryArrayToString` could be simpler and only support Uint8Array? e.g.

```suggestion
  if (ArrayBuffer.isView(body)) {
    if (body instanceof Uint8Array) {
      return { body: binaryArrayToString(body) };
    } else {
      return { body: JSON.stringify(body) };
    }
  }
```

Reply:
I think this makes sense! Thanks for the suggestion